sysvabi64/sysvabi64.rst

```diff
@@ -916,10 +923,657 @@ These follow the AAELF64_ base definition from Arm.
+in a pointer. Appropriate code is used to load the value of the
+pointer into a register, as for other data, and then an indirect
+branch instruction is used with the register as an operand to the
+instruction.
+
+In statically linked code the address of a function is always its real
+address. However in dynamically linked environments references to the
+address of a function from an executable file and its shared objects
+might not resolve to the same value.
+
+References from within shared objects to a function address will
+normally be resolved by the dynamic linker to the virtual address of
+the function itself.
+
+References from within a non-position-independent executable file to a
+function address defined in a shared object will normally be resolved
+by the static linker to the address of the `Procedure Linkage Table`_
+(PLT) entry for that function within the executable file. If the
+address is in a writeable location the linker may use a dynamic
+relocation instead.
+
+When both shared objects and a non-position-independent executable use
+the address of a function both the shared library and the application
+must use the same address for the functions in order for address
+comparisons to work as expected. If the static linker uses a PLT entry
+for a function as the address for a function then it must place the
+address of the PLT for the function in the dynamic symbols table entry
+for the function. This will result in a dynamic symbol table entry
+with a section index of ``SHN_UNDEF``, a type ``STT_FUNC`` and a
+non-zero ``st_value``. A reference to this symbol will be resolved by
+the dynamic linker to the address PLT for the function in the
+executable.
+
+The ``R_<CLS>_JUMP_SLOT`` relocations defined in AAELF64_ are
+associated with PLT entries. These entries are used for direct
+function calls rather than for references to function addresses. These
+relocations do not use the special symbol value described
+above. Otherwise a very tight endless loop would result.
+
+In a position-independent executable all non-local function addresses
+are accessed via the GOT so no indirection via the PLT entry is
+necessary.
+
+Shared objects may implement direct function calls to non-pre-emptable
+symbols using a direct branch instruction. A static linker will
+consider a symbol not pre-emptable if:
+
+ * The symbol has ``STB_LOCAL`` binding.
+
+ * The symbol has a symbol visibility that is not ``STV_DEFAULT``.
+
+ * The symbol has been given the ``local`` symbol version, defined in
+   SYM-VER_.
+
+ * The static linker adds the dynamic tag ``DT_FLAGS`` with the
+   ``DF_SYMBOLIC`` flag set. This makes all symbols non-pre-emptable.
+
+ * Some other implementation defined linker feature such as
+   ``--dynamic-list`` is used.
+
+Shared objects must indirect all function calls to pre-emptable
+symbols must be indirected through the static linker created Procedure
+Linkage Table.
+
```

Comment:
Phrasing -- "must indirect all function calls to pre-emptable symbols must be indirected".

Reply:
Ack have fixed.